### PR TITLE
Create new KeyEventsCards for carousel 

### PR DESCRIPTION
**Pull request by joecowton1**
## What does this change?

This PR is a WIP which creates new individual Key Event carousel cards. The styling will need refining as designs are completed but hopefully this gives something to work with.
## Screenshots

Screenshot 2022-04-21 at 21 20 24

### CODE REVIEW

File: dotcom-rendering/src/web/components/KeyEventsCards.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	return {
+		design: ArticleDesign.Standard,
+		display: ArticleDisplay.Standard,
+		theme,
+	};
+};
+
+const format = getFormat(ArticlePillar.News);
+
+const events: KeyEvent[] = [
```
Comment (abeddow91) [inline dotcom-rendering/src/web/components/KeyEventsCards.stories.tsx:25]: Non blocking but maybe we could move this into the fixtures?

Comment (abeddow91) [inline dotcom-rendering/src/web/components/KeyEventsCards.stories.tsx:25]: Non blocking but maybe we could move this into the fixtures?

File: dotcom-rendering/src/web/components/KeyEventsCards.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
+import { css } from '@emotion/react';
+import { textSans, neutral, space, from } from '@guardian/source-foundations';
+import { Link } from '@guardian/source-react-components';
+import { timeAgo } from '@guardian/libs';
+import { decidePalette } from '../lib/decidePalette';
+
+export interface KeyEvent {
```
Comment (abeddow91) [inline dotcom-rendering/src/web/components/KeyEventsCards.tsx:7]: This is a different model than what a key event is - they are just of type Block. I wonder if we should pass in key events as blocks here and move the logic that determines what's a date, text, url etc into this component?

File: dotcom-rendering/src/web/components/KeyEventsCards.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+);
+
+const MultipleCards = () => (
+	<>
+		<Cards theme={ArticlePillar.News} count={7} />
+		<Cards theme={ArticlePillar.Culture} count={7} />
+		<Cards theme={ArticlePillar.Lifestyle} count={7} />
+		<Cards theme={ArticlePillar.Sport} count={7} />
+		<Cards theme={ArticlePillar.Culture} count={7} />
+		<Cards theme={ArticlePillar.Opinion} count={1} />
```
Comment (marjisound) [inline dotcom-rendering/src/web/components/KeyEventsCards.stories.tsx:117]: Was this not supposed to be `count={7}` ?

Comment (abeddow91) [inline dotcom-rendering/src/web/components/KeyEventsCards.stories.tsx:117]: Count looks wrong here - guessing it should be 7?

Comment (abeddow91) [inline dotcom-rendering/src/web/components/KeyEventsCards.stories.tsx:117]: Count looks wrong here - guessing it should be 7?

**Review by abeddow91 — COMMENTED** [review]
Nice one! A few (non-blocking) comments but looks great :)

**Comment by joecowton1** [comment]
@abeddow91 @marjisound thanks for the comments! Can one of you please finish this one off in my absence?

File: dotcom-rendering/src/web/components/KeyEventsCards.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+	}
+`;
+
+const timeStyles = css`
+	${textSans.xsmall({ fontWeight: 'bold', lineHeight: 'tight' })};
+	color: ${neutral[7]};
+	display: block;
+`;
+
+export const KeyEventsCard = ({ text, date, url, format }: KeyEvent) => {
```
Comment (oliverlloyd) [inline dotcom-rendering/src/web/components/KeyEventsCards.tsx:84]: I think if the export is `KeyEventCard` and it's only exporting one thing then the filename should also be `KeyEventCard.tsx`?

**Comment by oliverlloyd** [comment]
I think the hover state needs adjusting? Maybe we can achieve this by lifting the link up?

File: dotcom-rendering/src/web/components/KeyEventsCards.tsx (hunk 4 above)
Comment (abeddow91) [reply dotcom-rendering/src/web/components/KeyEventsCards.tsx:84]: This component handles one key event so `KeyEventCard` feels right to me.

**Comment by abeddow91** [comment]
> I think the hover state needs adjusting? Maybe we can achieve this by lifting the link up? 

@oliverlloyd what would you expect instead here? I've taken a quick look at the existing implementation on DCR and I *think* these look the same. Maybe this is one for @benwuersching

**Comment by oliverlloyd** [comment]
> What would you expect instead here?

The hover state and the link itself (where the mouse pointer changes) should be aligned

**Comment by paperboyo** [comment]
Lovely. Is the line height a tad too loose? Toulouse.

**Comment by ajbreuer** [comment]
> Lovely. Is the line height a tad too loose? Toulouse.

@benwuersching @HarryFischer Can you check this? :)

**Comment by benwuersching** [comment]
Regarding line-height: please make it 135% (it's currently more than 160%). @abeddow91 @ajbreuer

**Review by oliverlloyd — APPROVED** [review]
Great work!

**Comment by joecowton1** [comment]
@abeddow91 thanks for seeing this through Anna! I've pushed a fix for the line-height so hopefully this is good to go now?

**Comment by alinaboghiu** [comment]
Yep, line height was the last thing as far as I know so good to merge.

**Pull request merged**